osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/deploymentspec/OsSvaServerCreateTask.java

```diff
@@ -160,41 +156,44 @@ public void executeTransaction(EntityManager em) throws Exception {
                         createdServer.getEgressInspectionMacAddr());
 
-                String portGroupId = ds.getPortGroupId();
-                boolean pgAlreadyCreatedByOther = (portGroupId != null);
+                if (this.apiFactoryService.supportsNeutronSFC(this.dai.getVirtualSystem())) {
+                    // In case of neutron SFC, port group id needs to be used when registering and updated in the DS
+                    String portGroupId = ds.getPortGroupId();
+                    boolean pgAlreadyCreatedByOther = (portGroupId != null);
+
+                    Element element = controller
+                            .registerInspectionPort(new DefaultInspectionPort(ingressPort, egressPort, null, portGroupId));
+
+                    portGroupId = element.getParentId();
 
-                if (this.apiFactoryService.supportsPortGroup(this.dai.getVirtualSystem())) {
+                    this.log.info(String.format("Setting port_group_id to %s on DAI %s (id %d) for Deployment Spec %s (id: %d)",
+                            portGroupId, this.dai.getName(), this.dai.getId(), ds.getName(), ds.getId()));
+
+                    if (!pgAlreadyCreatedByOther) {
+                        ds = em.find(DeploymentSpec.class, ds.getId());
+                        ds.setPortGroupId(portGroupId);
+                        OSCEntityManager.update(em, ds, this.txBroadcastUtil);
+                    }
+
+                } else if (this.apiFactoryService.supportsPortGroup(this.dai.getVirtualSystem())) {
                     String domainId = OpenstackUtil.extractDomainId(ds.getProjectId(),
                                                     ds.getProjectName(),
                                                     ds.getVirtualSystem().getVirtualizationConnector(),
                                                     Arrays.asList(ingressPort));
 
                     ingressPort.setParentId(domainId);
                     egressPort.setParentId(domainId);
-
-                    portGroupId = ds.getPortGroupId();
                 }
 
                 //Element object in DefaultInspectionport is not used at this point, hence null
-                Element element = controller.registerInspectionPort(
-                        new DefaultInspectionPort(ingressPort, egressPort, null, portGroupId));
-                portGroupId = element.getParentId();
-                this.log.info(String.format("Setting port_group_id to %s on DAI %s (id %d) for Deployment Spec %s (id: %d)",
-                        portGroupId, this.dai.getName(), this.dai.getId(), ds.getName(), ds.getId()));
-
-                if (!pgAlreadyCreatedByOther) {
-                    ds = em.find(DeploymentSpec.class, ds.getId());
-                    ds.setPortGroupId(portGroupId);
-                    OSCEntityManager.update(em, ds, this.txBroadcastUtil);
-                }
-            } finally {
-                controller.close();
+                controller.registerInspectionPort(
```

Comment:
This happens whether or not the call on line 170 happens.

Reply:
addressed